infra/buildtool/CMakeLists.txt

```diff
@@ -20,13 +20,26 @@ endmacro(nnas_include)
 
 nnas_include(OptionTools)
 envoption(EXTERNAL_DOWNLOAD_SERVER "https://github.com")
+# Sync with traditional external download tool: server login
+# TODO Migrate external download tool
+envoption(EXTERNAL_SERVER_USERPWD "")
+if(EXTERNAL_SERVER_USERPWD)
+  string(REGEX MATCH "^[^:]*" EXTERNAL_SERVER_USERNAME ${EXTERNAL_SERVER_USERPWD})
+  string(REGEX MATCH "[^:]*$" EXTERNAL_SERVER_PASSWORD ${EXTERNAL_SERVER_USERPWD})
```

Comment:
don't know much about this but this is to cut `USER`, `PWD` from `USER:PWD`  ?

Reply:
Right.